packages/blocks/src/api/raw-handling/phrasing-content-reducer.js

```diff
@@ -56,5 +56,25 @@ export default function phrasingContentReducer( node, doc ) {
 			node.removeAttribute( 'target' );
 			node.removeAttribute( 'rel' );
 		}
+
+		// Saves anchor elements name attribute as id
+		if ( node.name && ! node.id ) {
+			node.id = node.name;
+		}
+		// Removes obsolete name attribute in any case
+		node.removeAttribute( 'name' );
+
+		// Keeps id only if there is an internal link pointing to it
+		if (
+			node.id &&
+			! node.ownerDocument.querySelector( `[href="#${ node.id }"]` )
+		) {
+			node.removeAttribute( 'id' );
+		}
+
+		// Remove anchor tag if no href nor id
+		if ( ! node.id && ! node.href ) {
```

Comment:
Hm, shouldn't this only unwrap only if there's no link _elsewhere_? Why would a node have both an id and href?

Also, I'm not sure if we should concern ourselves with removing nodes here. That's something the `removeInvalidHTML` filter will (or should) take care of.

Reply:
Understood. I removed this check as well.